### PR TITLE
koji_promote adds koji_parent_id to build metadata additional metadata

### PR DESCRIPTION
**Pull request by rcerven**
Signed-off-by: Robert Cerven <rcerven@redhat.com>

### CODE REVIEW

File: tests/plugins/test_koji_promote.py [hunk 1]
```diff
@@ -1364,3 +1365,44 @@ def test_koji_promote_set_media_types(self, tmpdir, os_env, expect_result):
...
+    ])
+    def test_koji_promote_set_parent_id(self, tmpdir, os_env, parent_output, expect_result):
+        session = MockedClientSession('')
+        tasker, workflow = mock_environment(tmpdir,
+                                            name='ns/name',
+                                            version='1.0',
+                                            release='1',
+                                            session=session)
+
+        if parent_output:
```
Comment (vrutkovs) [inline tests/plugins/test_koji_promote.py]: Needs a test when `koji_parent` didn't run at all

File: tests/plugins/test_koji_promote.py [hunk 2]
```diff
@@ -1364,3 +1365,44 @@ def test_koji_promote_set_media_types(self, tmpdir, os_env, expect_result):
             assert sorted(image['media_types']) == sorted(expect_result)
         else:
             assert 'media_types' not in image.keys()
+
+    @pytest.mark.parametrize(('parent_output', 'expect_result'), [
```
Comment (vrutkovs) [inline tests/plugins/test_koji_promote.py]: nitpick: `expected_result`?

Comment (rcerven) [reply tests/plugins/test_koji_promote.py]: ok

File: tests/plugins/test_koji_promote.py (hunk 1 above)
Comment (rcerven) [reply tests/plugins/test_koji_promote.py]: I am mocking what is in workflow, and runner is running only koji_promote

so how such a test has value?

**Comment by rcerven** [comment]
commit message changed

**Comment by rcerven** [comment]
smoketests fail because of https://github.com/release-engineering/dockpulp/pull/147#issuecomment-332434655

File: tests/plugins/test_koji_promote.py (hunk 1 above)
Comment (lcarva) [reply tests/plugins/test_koji_promote.py]: Having a use case where `PLUGIN_KOJI_PARENT_KEY` key is not set at all in `prebuild_results` would cover that.

Comment (rcerven) [reply tests/plugins/test_koji_promote.py]: added

**Comment by vrutkovs** [comment]
Release notes updated

**Pull request merged**